src/utils/shadowBox.js

```diff
@@ -0,0 +1,188 @@
+import * as twgl from 'twgl.js';
+import { gl } from '../constants';
+import { deg2rad } from './math';
+import { Matrix4, Quaternion, Vector3 } from 'three';
+
+const m4 = twgl.m4;
+
+function aspectRatio() {
+  return gl.canvas.width / gl.canvas.height;
+}
+
+// vector representing the up direction
+const UP = new Vector3(0, 1, 0);
+
+// vector representing the forward direction
+const FORWARD = new Vector3(0, 0, -1);
+
+// value representing the maximum shadow distance
+const SHADOW_DISTANCE = 2000;
+
+// helps prevent weird times where shadows dont show up properly
+const DEPTH_OFFSET = 35;
+
+class ShadowBox {
+  /**
+   *
+   * @typedef {import('../camera').default} Camera
+   * @param {import('twgl.js').m4} lightViewMat
+   * @param {Camera} camera
+   */
+  constructor(lightViewMat, camera) {
+    this.lightViewMat = lightViewMat;
+    this.camera = camera;
+    this.calculateWidthsAndHeights();
+
+    // for the purposes of types and autocomplete
+    this.maxX = 0;
+    this.minX = 0;
+    this.maxY = 0;
+    this.minY = 0;
+    this.maxZ = 0;
+    this.minZ = 0;
+  }
+
+  update() {
+    const camRot = this.calculateCameraRotationMatrix();
+    const forwardVec = FORWARD.clone().applyMatrix4(camRot);
+
+    const toFar = forwardVec.clone().multiplyScalar(SHADOW_DISTANCE);
+    const toNear = forwardVec.clone().multiplyScalar(nearPlane());
```

Comment:
Was this line meant to be something else? `nearPlane` is defined anymore

Reply:
Fixed